### DISCUSSION TITLE
Does Aspire.Confluent.Kafka support distributed tracing?

### DISCUSSION THREAD

**Discussion by duskembayev** [microsoft/aspire#5629] [Q&A] [answered]
Hello!
I am using Aspire with Kafka and have found metrics, but I haven't come across any functionality related to tracing. Does Aspire.Confluent.Kafka support distributed tracing?

**Comment by davidfowl** [comment]
I don't think the underlying client library supports distributed tracing 😢. 

cc @eerhardt @g7ed6e

**Reply by eerhardt** [reply, marked as answer]
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1493/ added an OTel-contrib library to add distributed tracing for Kafka.

However, that library hasn't been published to nuget.org yet.  https://www.nuget.org/packages/OpenTelemetry.Instrumentation.ConfluentKafka

@g7ed6e says in https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1493#issuecomment-2273607331:

>  However this is not ready yet, there's still some work before publishing.

You can always find out what telemetry targets an Aspire integration library supports by looking at:

https://github.com/dotnet/aspire/blob/main/src/Components/Telemetry.md
